lib/propolis-client/src/instance_spec/devices.rs

```diff
@@ -268,6 +268,43 @@ impl MigrationElement for PciPciBridge {
     }
 }
 
+#[cfg(feature = "falcon")]
+#[derive(Clone, Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SoftNpuPciPort {
+    /// The PCI path at which to attach the guest to this port.
+    pub pci_path: PciPath,
+}
+
+#[cfg(feature = "falcon")]
+#[derive(Clone, Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SoftNpuPort {
+    /// The name of the SoftNpu port.
+    pub name: String,
+
+    /// The name of the associated VNIC.
```

Comment:
Minor nit: if this VNIC is similar to a `NetworkDevice` VNIC, i.e. it's some sort of host resource that's configured/managed outside of Propolis, then strictly speaking it should be defined over in the backend spec (and then this element can refer to that backend element by name the way a `NetworkDevice` does). I'm not sure if the existing virtio backend is fit for purpose here or if you'd need to define another backend type.

Reply:
I think this would be a different backend type than virtio. Virtio plugs into the VNIC with a viona data path, SoftNpu ports use DLPI.

Reply:
This field and the machinery around it have been updated to used a registered backend device.